idea/idea-frontend-fir/src/org/jetbrains/kotlin/idea/frontend/api/fir/components/KtFirExpressionTypeProvider.kt

```diff
@@ -35,14 +35,23 @@ internal class KtFirExpressionTypeProvider(
     }
 
     override fun getKtExpressionType(expression: KtExpression): KtType = withValidityAssertion {
-        when (val fir = expression.getOrBuildFir(firResolveState)) {
+        when (val fir = expression.unwrap().getOrBuildFir(firResolveState)) {
             is FirExpression -> fir.typeRef.coneType.asKtType()
             is FirNamedReference -> fir.getReferencedElementType().asKtType()
             is FirStatement -> with(analysisSession) { builtinTypes.UNIT }
             else -> error("Unexpected ${fir::class}")
         }
     }
 
+    private fun KtExpression.unwrap(): KtExpression {
+        return when (this) {
+            is KtLabeledExpression -> baseExpression?.unwrap()
+            is KtAnnotatedExpression -> baseExpression?.unwrap()
+            is KtObjectLiteralExpression -> objectDeclaration
+            else -> null
```

Comment:
Maybe handling `KtParenthesizedExpression` as well?

Reply:
I guess that would not hurt. But `getOrBuildFir` is returning the expected result without special handling here. Let me add more test cases covering that.

Reply:
You may want to recursively unwrap for parentheses. See `PsiElement?.unwrapParenthesesLabelsAndAnnotations` in `compiler/psi/src/org/jetbrains/kotlin/psi/psiUtil/psiUtils.kt`. Or maybe you can modify and use it to unwrap those, and add your own unwrapping for `KtObjectLiteralExpression`

Reply:
I am not sure it's desirable to make org.jetbrains.kotlin.psi.psiUtil.PsiUtilsKt#unwrapParenthesesLabelsAndAnnotations unwrap `KtObjectLiteralExpression` since it seems to be a different kind of wrap comparing with parentheses, annotation, and labels.

Regarding multiple parentheses, it's actually handled correctly by underlying FIR in this case and there is nothing we need to do additionally to handle it.